libs/unity/library/Runtime/Scripts/Media/MediaLine.cs

```diff
@@ -322,26 +319,28 @@ internal void DetachTrack()
 
         internal void UpdateReceiver()
         {
-            var transceiver = Transceiver;
-            Debug.Assert(transceiver != null);
-            bool wantsRecv = (Receiver != null);
-            bool wasReceiving = (_pairedReceiver != null);
-            bool isReceiving = (transceiver.RemoteTrack != null);
-            // Note the extra "isReceiving" check, which ensures that when the remote track was
-            // just removed by OnUnpaired(RemoteTrack) from the TrackRemoved event then it is not
-            // immediately re-added by mistake.
-            var receiverInternal = (_receiver as IMediaReceiverInternal);
-            if (wantsRecv && isReceiving && !wasReceiving)
+            if (_receiver != null)
```

Comment:
This seems to break the logic : if a user removes its receiver object from the MediaLine of the peer connection, we should clear the receiver track no? Or is it precisely that we cannot remove a receiver track without renegotiation, so changing the Receiver on the MediaLine should not have any effect until it's renegotiated?

Reply:
> Or is it precisely that we cannot remove a receiver track without renegotiation, so changing the Receiver on the MediaLine should not have any effect until it's renegotiated?

This seems to be correct - in any case sounds unrelated, if a user removes its receiver object from the MediaLine they won't go through this method, even before my change.

Re this specific change, this has the only possible effect of calling methods on `_receiver` so it makes sense to skip if there is no receiver. I guess the name `UpdateReceiver` is ambiguous, will rename to `UpdateReceiverPairingIfNeeded`.

Reply:
This has been refactored in a different PR.